tools/tck-regression/src/test/java/org/opencypher/tools/tck/regression/TckResultsComparatorTest.java

```diff
@@ -0,0 +1,255 @@
+        String path = temp.newFolder().getAbsolutePath();
+
+        assertThat(comparatorException(test4, "--compareTo", test0, "--output", path),
+            containsString("has failures"));
+
+        String content = read(path, "regression.html");
+
+        assertThat(content, containsString("TCK coverage degraded"));
+        assertThat(content, containsString(failing(1)));
+    }
+
+    @Test
+    public void testDiffSize() throws Exception {
+        String path = temp.newFolder().getAbsolutePath();
+
+        assertThat(comparatorException(test1, "--compareTo", test2, "--output", path),
+            containsString("size changed"));
+
+        assertThat(comparatorException(test2, "--compareTo", test1, "--output", path),
+            containsString("size changed"));
+
+        assertThat(new File(path, "regression.html").exists(), is(false));
+    }
+
+
+    @Test
+    public void testVerifyOnlyFailed() throws Exception {
+        String path = temp.newFolder().getAbsolutePath();
+
+        assertThat(comparatorException(test0, "--verifyFailures", test1, "--output", path),
+            containsString("only failed scenarios"));
+
+        assertThat(new File(path, "regression.html").exists(), is(false));
+    }
+
+    @Test
+    public void testVerifyImprove() throws Exception {
+        String path = temp.newFolder().getAbsolutePath();
+
+        assertThat(comparatorException(test0, "--verifyFailures", test2, "--output", path),
+            containsString("does not match"));
+
+        String content = read(path, "regression.html");
+
+        assertThat(content, containsString(passing(2)));
+    }
+
+    @Test
+    public void testVerifyNoChange() throws Exception {
+        String path = temp.newFolder().getAbsolutePath();
+
+        comparator(test2, "--verifyFailures", test2, "--output", path);
+
+        String content = read(path, "regression.html");
+
+        assertThat(content, containsString("Nothing changed"));
+    }
+
+    @Test
+    public void testVerifyDegrade() throws Exception {
+        String path = temp.newFolder().getAbsolutePath();
+
+        assertThat(comparatorException(test3, "--verifyFailures", test2, "--output", path),
+            containsString("does not match"));
+
+        String content = read(path, "regression.html");
+
+        assertThat(content, containsString(failing(1)));
+    }
+
+    @Test
+    public void testConvertCsv() throws Exception {
+        String path = temp.newFolder().getAbsolutePath();
+
+        comparator(test4, "--saveCsv", "--output", path);
+
+        String actual = read(path, "base.csv").replaceAll("\r\n", "\n");
+
+        String expected = read("", test1);
+
+        assertThat(actual, equalTo(expected));
+    }
+
+
+    private void comparator(String... args) throws Exception {
+        TckResultsComparator.main(args);
+    }
+
+    private String comparatorException(String... args) {
+        try {
+            comparator(args);
+            return null;
```

Comment:
I have not used `ExpectedException` because additional verifications need to be done after exception is thrown. Added failure if exception is not thrown.